conbench/app/benchmarks.py

```diff
@@ -126,17 +131,39 @@ def _augment(self, run):
 
         # Note(JP): `run["commit"]["timestamp"]` can be `None`, see
         # https://github.com/conbench/conbench/pull/651
+        # Does run["commit"] every result in KeyError?
         self._display_time(run["commit"], "timestamp")
         repository = run["commit"]["repository"]
         repository_name = repository
+
         if "github.com/" in repository:
             repository_name = repository.split("github.com/")[1]
+
         run["display_name"] = ""
         if run["name"]:
             run["display_name"] = run["name"].split(":", 1)[0]
+
         run["commit"]["display_repository"] = repository_name
+
+        # Note(JP): does run["commit"]["message"] ever result in KeyError?
+        # I think `display_message()` may be thought of constructing the text
+        # for a URL. But.... shrug. This needs consolidation.
         commit_message = display_message(run["commit"]["message"])
+
+        # Note(JP): run.commit.url and run.commit.display_message seem to be
+        # consumed in the HTML template. Here I am a little lost about the
+        # guarantees -- are they always available? We need to resolve this with
+        # proper type annotations and schemata. Until then, do poor-man's
+        # validation to prevent AttributeError and KeyError.
+        # display_message really seems to be name of the link
         run["commit"]["display_message"] = commit_message
+        if not commit_message:  # None or empty string
+            if run["commit"].get("url"):  # be real conservative
+                run["commit"]["display_message"] = run["commit"].get("url")
+            else:
+                # let the template consume these two keys.
+                run["commit"]["url"] = "#"
+                run["commit"]["display_message"] = "not commit info"
```

Comment:
Soooo looking forward to passing typed structures around. I consider this here to be very temporary now.